src/expandableCalendar/WeekCalendar/presenter.spec.js

```diff
@@ -0,0 +1,359 @@
+      const context = makeContext();
+      const firstDay = 0;
+      const presenter = makeUUT();
+      const props = {current: '2021-01-31', context, firstDay};
+      expect(presenter.getDate(props, 0)).toEqual('2021-01-31');
+
+      expect(presenter.getDate(props, -1)).toEqual('2021-01-24');
+      expect(presenter.getDate(props, -2)).toEqual('2021-01-17');
+      expect(presenter.getDate(props, -3)).toEqual('2021-01-10');
+    });
+
+    it('Expect to get a week after current week by passing positive index', () => {
+      const presenter = makeUUT();
+      const props = makeProps('2021-01-31');
+      expect(presenter.getDate(props, 0)).toEqual('2021-01-31');
+
+      expect(presenter.getDate(props, 1)).toEqual('2021-02-07');
+      expect(presenter.getDate(props, 2)).toEqual('2021-02-14');
+      expect(presenter.getDate(props, 3)).toEqual('2021-02-21');
+    });
+
+    it('Expect to be able to get past year', () => {
+      const presenter = makeUUT();
+      const props = makeProps('2021-01-07');
+      expect(presenter.getDate(props, -1)).toEqual('2020-12-27');
+    });
+
+    it('Expect to be able to get next year', () => {
+      const props = makeProps('2021-12-30');
+      const presenter = makeUUT();
+      expect(presenter.getDate(props, 1)).toEqual('2022-01-02');
+    });
+  });
+
+  describe('getMarkedDates test', () => {
+    const markedDates = {
+      markedDates: {'2021-01-10': {marked: true}, '2021-01-11': {endingDay: false}, '2021-01-12': {endingDay: true}}
+    };
+
+    it('Expect getMarkedDates to be current date, when passing context and markedDates are null', () => {
+      const context = {date: '2021-01-12'};
+      const {getMarkedDates} = makeUUT();
+      expect(getMarkedDates(context, undefined)).toEqual({'2021-01-12': {selected: true}});
+    });
+
+    it('Expect getMarkedDate to return the passed markedDates value when available', () => {
+      const context = {date: '2021-01-11'};
+      const {getMarkedDates} = makeUUT();
+      expect(getMarkedDates(context, markedDates)).toEqual({
+        '2021-01-11': {selected: true},
+        ...markedDates
+      });
+    });
+
+    it('Expect getMarkedDate to return the passed markedDates value when available', () => {
+      const markedDates = {markedDates: {'2021-01-11': {endingDay: false}, '2021-01-12': {endingDay: true}}};
+      const context = {date: '2021-01-11'};
+      const {getMarkedDates} = makeUUT();
+      expect(getMarkedDates(context, markedDates)).toEqual({
+        '2021-01-11': {selected: true},
+        markedDates: {'2021-01-11': {endingDay: false}, '2021-01-12': {endingDay: true}}
+      });
+    });
+  });
+
+  describe('Event - onDatPressed', () => {
+    it('onDayPressed', () => {
+      const setDate = jest.fn();
+      const context = {date: '2021-02-02', setDate};
+      const {onDayPressed} = makeUUT();
+
+      onDayPressed(context, {dateString: '2021-01-12'});
+      expect(setDate).toBeCalled();
```

Comment:
'setDate' is on the ContextProvider. Shouldn't we check that the `this.props.context.date' has been set to the given date instead?

Reply:
totally right

Reply:
Can't you access the context from the props?

Reply:
I can check if it's called with the right value, maybe it's better:
`expect(setDate).toBeCalledWith('2021-01-22', UPDATE_SOURCES.DAY_PRESS);`
I'm not sure if checking the parent state here is the right place to do so, WDYT?